.github/workflows/ci.yml

```diff
@@ -33,7 +31,29 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  changed:
+    name: Get changed files
+    runs-on: macos-14
+    outputs:
+      should_skip: ${{ steps.changed-files.outputs.only_changed == 'true' }}
+
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
We also need to copy the fetch-depth option from below so the action can compare the change from previous commits. `actions/checkout` defaults to 1.

We can then remove the option in the below `actions/checkout` too because we only need the depth for the `changed-files` action.